### ISSUE TITLE
`p len(string(data))` != `p len(data)`

### ISSUE THREAD

**Issue by extemporalgenome** [go-delve/delve#3539]
Please answer the following before submitting your issue:

1. What version of Delve are you using (`dlv version`)?
```
Delve Debugger
Version: 1.21.0
Build: $Id: fec0d226b2c2cce1567d5f59169660cf61dc1efe $
```

3. What version of Go are you using? (`go version`)?
`go version go1.21.0 darwin/amd64`

4. What operating system and processor architecture are you using?

darwin/amd64

5. What did you do?
Print the backing array contents of a large-ish []byte as a string.

6. What did you expect to see?

```
(dlv) p len(data)
1202
(dlv) p len(string(data))
1202
```

7. What did you see instead?

```
(dlv) p len(data)
1202
(dlv) p len(string(data))
64
```

**Comment by extemporalgenome** [comment]
It appears that when given a []byte, `max-array-values` must be at least as large as `max-string-len` for it to be respected when performing this command: `p string(theByteSlice)`.

My expectation would be that `max-array-values` is only utilized at the time of presentation, and that `p string(theByteSlice)` would ignore the value of `max-array-values` entirely, convert (or interpret) the entire slice as a string, and then use the value of `max-string-len` to configure the presentation of the result.

**Comment by aarzilli** [comment]
This is essentially by design. The various limits (max-array-values, max-string-length, &c) don't apply to presentation but to how much memory is read. This is so that we don't try to read infinite amounts of memory even in presence of corrupted values.

Converting a byte array to a string involves reading the byte array and reinterpreting its contents, thus the limits on how many array elements we read are applied. We could fix this particular instance of the problem but it would just push the problem one level up: `p len([]rune(string(byteslice)))`

**Comment by derekparker** [comment]
I wonder if there could be a way to optimize for this in our evaluator, because technically in this case we don't need to load the value of the string, we really just read the length field from the structs and compare those.

**Comment by aarzilli** [comment]
Closing, fixed by 57dad9342a9443e298d43fcc653475408402131e

**Comment by extemporalgenome** [comment]
That's fantastic! Thanks!
